session_pool.go

```diff
@@ -410,33 +410,41 @@ func (pool *SessionPool) getSessionFromIdle() (*pureSession, error) {
 //  1. the current session is invalid, and then get idle session from pool
 //  2. the connection is invalid, and do not logout, just retry execution
 func (pool *SessionPool) executeWithRetry(session *pureSession,
-	f func(*pureSession) (*ResultSet, error), retry int) (*ResultSet, error) {
-	return pool.executeWithRetryLimit(session, f, 0, retry)
+	f func(*pureSession) (*ResultSet, error), sessionRetry int, errRetry int) (*ResultSet, error) {
+	return pool.executeWithRetryLimit(session, f, 0, sessionRetry, 0, errRetry)
 }
 
 func (pool *SessionPool) executeWithRetryLimit(session *pureSession,
-	f func(*pureSession) (*ResultSet, error), retryTimes, retryLimit int) (*ResultSet, error) {
-	rs, err := f(session)
-	if retryTimes >= retryLimit {
-		return rs, err
-	}
+	fn func(*pureSession) (*ResultSet, error),
+	sessionRetryTimes, sessionRetryLimit int,
+	errRetryTimes, errRetryLimit int,
+) (*ResultSet, error) {
+	rs, err := fn(session)
+
 	if err == nil {
 		if rs.GetErrorCode() != ErrorCode_E_SESSION_INVALID {
 			return rs, nil
-		} else {
-			if err := pool.retryStrategySessionInvalid(session); err != nil {
-				pool.log.Error(fmt.Sprintf("cannot retry when session is invalid, error: %s", err.Error()))
-				return nil, err
-			}
 		}
+		if sessionRetryTimes >= sessionRetryLimit {
+			return rs, nil
+		}
+		pool.log.Info(fmt.Sprintf("retry %d times to execute the query for session invalid", sessionRetryTimes+1))
```

Comment:
The same. Suggest to move the variable to the end of the log, like
 
`retry to execute the query %d times for session invalid`